corehq/apps/hqcase/utils.py

```diff
@@ -255,3 +257,15 @@ def bulk_update_cases(domain, case_changes, device_id):
         case_block = ElementTree.tostring(case_block.as_xml())
         case_blocks.append(case_block)
     return submit_case_blocks(case_blocks, domain, device_id=device_id)
+
+
+def resync_case_to_es(domain, case):
+    if should_use_sql_backend(domain):
+        publish_case_saved(case)
+    else:
+        CommCareCase.get_db().save_doc(case._doc)  # don't just call save to avoid signals
+
+
+def resave_case(domain, case):
```

Comment:
Why is this method necessary?

Reply:
Agree could have just called `resync_case_to_es` but added this method as a single place where we want to do things when re-saving case. Also keeps the same vocab as resave_form.
Plus,
`resync_case_to_es` is one of the specific things but overall might want to club this with something like resave to UCRs in future.
Fine to remove it if you think it would never be needed.

Reply:
Fine with keeping the function name `resave_case` but I don't think we need the two separate methods. This will re-trigger any UCRs for this case as well since what this is doing is putting the case in the kafka queue again so it will get picked up by all pillows listening to the case topic.

Reply:
cool, renaming.